src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -653,6 +663,33 @@ public void finish() {
             return coverageDistribution;
         }
 
+        /** Emits a file with per base coverage if an output file has been set. */
+        private void emitPerBaseCoverageIfRequested() {
+            if (this.perBaseOutput == null) return;
+
+            final PrintWriter out = new PrintWriter(IOUtil.openFileForBufferedWriting(this.perBaseOutput));
+            out.println("chrom\tpos\ttarget\tcoverage");
+            for (final Map.Entry<Interval,Coverage> entry : this.coverageByTarget.entrySet()) {
+                final Interval interval = entry.getKey();
+                final String chrom = interval.getContig();
+                final int firstBase = interval.getStart();
+
+                final int[] cov = entry.getValue().getDepths();
+                for (int i = 0; i < cov.length; ++i) {
+                    out.print(chrom);
+                    out.print('\t');
+                    out.print(firstBase + i);
+                    out.print('\t');
+                    out.print(interval.getName());
+                    out.print('\t');
+                    out.print(cov[i]);
+                    out.println();
+                }
+            }
+
```

Comment:
I tned to leave a newline after closing a block, to provide some visual separation.